frontend/packages/ceph-storage-plugin/src/components/modals/advanced-kms-modal/advanced-kms-modal.tsx

```diff
@@ -0,0 +1,284 @@
+
+export const AdvancedKMSModal = withHandlePromise((props: AdvancedKMSModalProps) => {
+  const { close, cancel, errorMessage, inProgress, state, dispatch, mode } = props;
+  const { kms } = state;
+  const [backendPath, setBackendPath] = React.useState(kms.backend || '');
+  const [tlsName, setTLSName] = React.useState(kms.tls || '');
+  const [caCertificate, setCACertificate] = React.useState(kms.caCert?.stringData['ca.cert'] || '');
+  const [caCertificateFile, setCACertificateFile] = React.useState(kms.caCertFile || '');
+  const [clientCertificate, setClientCertificate] = React.useState(
+    kms.clientCert?.stringData['tls.cert'] || '',
+  );
+  const [clientCertificateFile, setClientCertificateFile] = React.useState(
+    kms.clientCertFile || '',
+  );
+  const [clientKey, setClientKey] = React.useState(kms.clientCert?.stringData['tls.key'] || '');
+  const [clientKeyFile, setClientKeyFile] = React.useState(kms.clientKeyFile || '');
+  const [providerNS, setProvideNS] = React.useState(kms.providerNamespace || '');
+  const [error, setError] = React.useState('');
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    const caSecret: SecretKind = {
+      apiVersion: SecretModel.apiVersion,
+      kind: SecretModel.kind,
+      metadata: {
+        generateName: 'ocs-kms-ca-',
+        namespace: CEPH_STORAGE_NAMESPACE,
+      },
+      stringData: {
+        'ca.cert': caCertificate,
+      },
+    };
+
+    const clientCertSecret: SecretKind = {
+      apiVersion: SecretModel.apiVersion,
+      kind: SecretModel.kind,
+      metadata: {
+        generateName: 'ocs-kms-client-cert-',
+        namespace: CEPH_STORAGE_NAMESPACE,
+      },
+      stringData: {
+        'tls.cert': clientCertificate,
+      },
+    };
+
+    const clientKeySecret: SecretKind = {
+      apiVersion: SecretModel.apiVersion,
+      kind: SecretModel.kind,
+      metadata: {
+        generateName: 'ocs-kms-client-key-',
+        namespace: CEPH_STORAGE_NAMESPACE,
+      },
+      stringData: {
+        'tls.key': clientKey,
+      },
+    };
+
+    const kmsAdvanced = {
+      ...kms,
+      backend: backendPath,
+      tls: tlsName,
+      providerNamespace: providerNS,
+    };
+
+    if (caCertificate && caCertificate !== '') {
+      kmsAdvanced.caCert = caSecret;
+      kmsAdvanced.caCertFile = caCertificateFile;
+    } else {
+      kmsAdvanced.caCert = null;
+      kmsAdvanced.caCertFile = '';
+    }
+
+    if (clientCertificate && clientCertificate !== '') {
+      kmsAdvanced.clientCert = clientCertSecret;
+      kmsAdvanced.clientCertFile = clientCertificateFile;
+    } else {
+      kmsAdvanced.clientCert = null;
+      kmsAdvanced.clientCertFile = '';
+    }
+
+    if (clientKey && clientCertificate !== '') {
+      kmsAdvanced.clientKey = clientKeySecret;
+      kmsAdvanced.clientKeyFile = clientKeyFile;
+    } else {
+      kmsAdvanced.clientKey = null;
+      kmsAdvanced.clientKeyFile = '';
+    }
```

Comment:
Does not setting value helps here ?
Because anyways you are setting state. I think you can avoid the guard here